datacube_stats/main.py

```diff
@@ -824,6 +851,14 @@ def __call__(self, index, sources_spec, date_ranges):
             # Each source may be masked by multiple masks
             tasks = {}
             for source_spec in sources_spec:
+                ep_range = time_period
```

Comment:
This computation is about 7 lines long and is repeated in this file 3 times, I can't tell just from looking at it if it's exactly the same computation or slightly different depending on the context. Please take it out into a function.